pkg/cloud/gcp/actuators/machine/reconciler.go

```diff
@@ -119,39 +119,70 @@ func (r *Reconciler) create() error {
 
 	operation, err := r.computeService.InstancesInsert(r.projectID, zone, instance)
 	if err != nil {
+		if reconcileWithCloudError := r.reconcileMachineWithCloudState(&v1beta1.GCPMachineProviderCondition{
+			Type:    v1beta1.MachineCreated,
+			Reason:  machineCreationFailedReason,
+			Message: err.Error(),
+			Status:  corev1.ConditionFalse,
+		}); reconcileWithCloudError != nil {
+			klog.Errorf("Failed to reconcile machine with cloud state: %v", reconcileWithCloudError)
+		}
 		return fmt.Errorf("failed to create instance via compute service: %v", err)
 	}
 	if op, err := r.waitUntilOperationCompleted(zone, operation.Name); err != nil {
+		if reconcileWithCloudError := r.reconcileMachineWithCloudState(&v1beta1.GCPMachineProviderCondition{
+			Type:    v1beta1.MachineCreated,
+			Reason:  machineCreationFailedReason,
+			Message: err.Error(),
+			Status:  corev1.ConditionFalse,
+		}); reconcileWithCloudError != nil {
+			klog.Errorf("Failed to reconcile machine with cloud state: %v", reconcileWithCloudError)
+		}
 		return fmt.Errorf("failed to wait for create operation via compute service. Operation status: %v. Error: %v", op, err)
 	}
-	return nil
+	return r.reconcileMachineWithCloudState(nil)
```

Comment:
This has an api call that could potentially fail, even though the instance was created (same with a timeout of waitUntilOperationCompleted above).  We need to add logic to check that a machine was already created (instance.insert), so we don't fail on trying to create it again, rinse repeat.

Reply:
If any of those fail then the machine controller triggers a new reconciling iteration, then:
- The machine controller calls Exists() the actuator returns true, then:
- The machine controller calls update() the actuator reconciles state with cloud
This is by design.